File: linjiashop-admin/src/main/java/com/ityu/linjiaadminapi/controller/system/UserController.java
package com.ityu.linjiaadminapi.controller.system;


import com.ityu.bean.constant.Const;
import com.ityu.bean.constant.factory.PageFactory;
import com.ityu.bean.constant.state.ManagerStatus;

import com.ityu.core.BussinessLog;

import com.ityu.bean.dto.UserDto;
import com.ityu.bean.entity.system.User;

import com.ityu.bean.enumeration.BizExceptionEnum;
import com.ityu.bean.exception.ApplicationException;
import com.ityu.core.factory.UserFactory;
import com.ityu.service.system.ManagerService;
import com.ityu.utils.BeanUtil;
import com.ityu.utils.RandomUtil;
import com.ityu.utils.StringUtil;
import com.ityu.utils.factory.Page;
import com.ityu.bean.vo.front.Rets;
import com.ityu.bean.vo.query.SearchFilter;
import com.ityu.warpper.UserWarpper;
import com.ityu.web.controller.BaseController;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;
import java.util.List;

/**
 * UserController
 *
 * @author enilu
 * @version 2018/9/15 0015
 */
@RestController
@RequestMapping("/api/user")
public class UserController extends BaseController {
    @Autowired
    private ManagerService managerService;
    @Autowired
    private BCryptPasswordEncoder bCryptPasswordEncoder;


    @RequestMapping(value = "/list", method = RequestMethod.GET)
    //@RequiresPermissions(value = {Permission.USER})
    public Object list(@RequestParam(required = false) String account,
                       @RequestParam(required = false) String name,
                       @RequestParam(required = false) Integer sex) {
        Page page = new PageFactory().defaultPage();
        if(StringUtil.isNotEmpty(name)){
            page.addFilter(SearchFilter.build("name", SearchFilter.Operator.LIKE, name));
        }
        if(StringUtil.isNotEmpty(account)){
            page.addFilter(SearchFilter.build("account", SearchFilter.Operator.LIKE, account));
        }
        page.addFilter( "status",SearchFilter.Operator.GT,0);
        page.addFilter("sex", sex);
        page = managerService.queryPage(page);
        List list = (List) new UserWarpper(BeanUtil.objectsToMaps(page.getRecords())).warp();
        page.setRecords(list);
        return Rets.success(page);
    }

    @RequestMapping(method = RequestMethod.POST)
    @BussinessLog(value = "编辑管理员", key = "name")
    // @RequiresPermissions(value = {Permission.USER_EDIT})
    public Object save(@Valid UserDto user) {
        if (user.getId() == null) {
            // 判断账号是否重复
            User theUser = managerService.findByAccount(user.getAccount());
            if (theUser != null) {
                throw new ApplicationException(BizExceptionEnum.USER_ALREADY_REG);
            }
            if(user.getPassword()==null){
                return Rets.failure("密码不能为空");
            }
            // 完善账号信息
            user.setSalt(RandomUtil.getRandomString(5));
            user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
            user.setStatus(ManagerStatus.OK.getCode());
            managerService.insert(UserFactory.createUser(user, new User()));
        } else {
            User oldUser = managerService.get(user.getId());
            managerService.update(UserFactory.updateUser(user, oldUser));
        }
        return Rets.success();
    }

    @BussinessLog(value = "删除账号", key = "userId")
    @RequestMapping(method = RequestMethod.DELETE)
    //@RequiresPermissions(value = {Permission.USER_DEL})
    public Object remove(@RequestParam Long userId) {
        if (userId == null) {
            throw new ApplicationException(BizExceptionEnum.REQUEST_NULL);
        }
        if (userId.intValue() <= 2) {
            return Rets.failure("不能删除初始用户");
        }
        User user = managerService.get(userId);
        user.setStatus(ManagerStatus.DELETED.getCode());
        managerService.update(user);
        return Rets.success();
    }

    @BussinessLog(value = "设置账号角色", key = "userId")
    @RequestMapping(value = "/setRole", method = RequestMethod.GET)
    //@RequiresPermissions(value = {Permission.USER_EDIT})
    public Object setRole(@RequestParam("userId") Long userId, @RequestParam("roleIds") String roleIds) {
        if (BeanUtil.isOneEmpty(userId, roleIds)) {
            throw new ApplicationException(BizExceptionEnum.REQUEST_NULL);
        }
        //不能修改超级管理员
        if (userId.intValue() == Const.ADMIN_ID.intValue()) {
            throw new ApplicationException(BizExceptionEnum.CANT_CHANGE_ADMIN);
        }
        User user = managerService.get(userId);
        user.setRoleid(roleIds);
        managerService.update(user);
        return Rets.success();
    }

    @BussinessLog(value = "冻结/解冻账号", key = "userId")
    @RequestMapping(value = "changeStatus", method = RequestMethod.GET)
    //@RequiresPermissions(value = {Permission.USER_EDIT})
    public Object changeStatus(@RequestParam Long userId) {
        if (userId == null) {
            throw new ApplicationException(BizExceptionEnum.REQUEST_NULL);
        }
        User user = managerService.get(userId);
        user.setStatus(user.getStatus().intValue() == ManagerStatus.OK.getCode() ? ManagerStatus.FREEZED.getCode() : ManagerStatus.OK.getCode());
        managerService.update(user);
        return Rets.success();
    }

}

File: sdf-api/src/main/java/com/ityu/sdf/api/controller/system/UserController.kt
package com.ityu.sdf.api.controller.system

import com.ityu.common.bean.constant.Const
import com.ityu.common.bean.constant.factory.PageFactory
import com.ityu.common.bean.constant.state.ManagerStatus
import com.ityu.common.bean.core.BussinessLog
import com.ityu.common.bean.entity.system.User
import com.ityu.common.bean.vo.front.Rets.success
import com.ityu.common.bean.vo.query.SearchFilter
import com.ityu.common.core.factory.UserFactory
import com.ityu.common.dto.UserDto
import com.ityu.common.exception.Asserts
import com.ityu.common.exception.ResultCode
import com.ityu.common.service.system.UserService
import com.ityu.common.utils.BeanUtil
import com.ityu.common.utils.CryptUtil
import com.ityu.common.utils.factory.Page
import com.ityu.common.warpper.UserWrapper
import com.ityu.sdf.api.utils.ApiConstants
import io.swagger.annotations.Api
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.web.bind.annotation.*
import jakarta.validation.Valid

/**
 * UserController
 *
 * @author enilu
 * @version 2018/9/15 0015
 */
@Api(tags = ["user"])
@Tag(name = "user", description = "用户模块")
@RestController
@RequestMapping("/user")
open class UserController(
    private var passwordEncoder: PasswordEncoder,
    private var userService: UserService,

    ) {
    @GetMapping(value = ["/list"])
    open fun list(
        @RequestParam(required = false) account: String?,
        @RequestParam(required = false) name: String?,
        @RequestParam(required = false) deptid: Long?,
        @RequestParam(required = false) phone: String?,
        @RequestParam(required = false) status: Int?,
        @RequestParam(required = false) sex: Int?
    ): Any {
        var page: Page<User?> = PageFactory<User>().defaultPage()
        page.addFilter("name", SearchFilter.Operator.LIKE, name)
        page.addFilter("account", SearchFilter.Operator.LIKE, account)
        page.addFilter("deptid", deptid)
        page.addFilter("phone", phone)
        page.addFilter("status", status)
        page.addFilter("status", SearchFilter.Operator.GT, 0)
        page.addFilter("sex", sex)
        page = userService.queryPage(page)
        val list = UserWrapper(BeanUtil.objectsToMaps(page.records)).warp() as List<User>
        page.setRecords(list)
        return success(page)
    }

    @BussinessLog(value = "重置密码", key = "userId")
    @PostMapping(value = ["resetPassword"])
    open fun resetPassword(userId: Long): Any {
        val user = userService.get(userId)
        user!!.password = passwordEncoder.encode("111111")
        userService.update(user)
        return success()
    }

    @PostMapping
    @BussinessLog(value = "编辑账号", key = "name")
    open fun save(@Valid @RequestBody user: UserDto): Any {
        if (user.id == null) {
            user.let {
                val theUser: User? = userService.findByAccountForLogin(it.account)
                if (theUser != null) {
                    Asserts.fail(ResultCode.USER_ALREADY_REG)
                }
                it.password = passwordEncoder.encode(CryptUtil.desEncrypt(user.password))
                it.status = ManagerStatus.OK.code
                userService.insert(UserFactory.createUser(user, User()))
            }

        } else {
            val oldUser: User = userService.get(user.id)!!
            userService.update(UserFactory.updateUser(user, oldUser))
        }
        return success()
    }


    @BussinessLog(value = "删除账号", key = "userId")
    @DeleteMapping
    open fun remove(@RequestParam userId: Long?): Any {
        if (userId == null) {
            Asserts.fail(ResultCode.REQUEST_NULL)
        }
        if (userId!!.toInt() <= 3) {
            Asserts.fail("不能删除初始用户")
        }
        val user = userService[userId]
        user!!.status = ManagerStatus.DELETED.code
        userService.update(user)
        return success()
    }

    @BussinessLog(value = "设置账号角色", key = "userId")
    @PostMapping(value = ["/setRole"])
    open fun setRole(@RequestParam("userId") userId: Long, @RequestParam("roleIds") roleIds: String?): Any {
        if (BeanUtil.isOneEmpty(userId, roleIds)) {
            Asserts.fail(ResultCode.REQUEST_NULL)
        }
        //不能修改超级管理员
        if (userId.toInt() == Const.ADMIN_ID) {
            Asserts.fail("不能修改超级管理员得角色")
        }
        val user = userService[userId]
        user!!.roleid = roleIds
        userService.update(user)
        return success()
    }

    @BussinessLog(value = "冻结/解冻账号", key = "userId")
    @GetMapping(value = ["changeStatus"])
    open fun changeStatus(@RequestParam userId: Long?): Any {
        if (userId == null) {
            Asserts.fail(ResultCode.REQUEST_NULL)
        }
        if (userId!!.toInt() <= 3) {
            Asserts.fail("不能冻结初始用户")
        }
        val user = userService[userId]
        user!!.status = (
                if (user!!.status === ManagerStatus.OK.code
                ) ManagerStatus.FREEZED.code else ManagerStatus.OK.code
                )
        userService.update(user)
        return success()
    }


}
